src/type-translator.ts

```diff
@@ -174,17 +180,38 @@ export class TypeTranslator {
     }
     const alias = this.symbolsToAliasedNames.get(symAlias);
     if (alias) return alias;
-    if (useFqn) {
-      const fqn = this.typeChecker.getFullyQualifiedName(sym);
-      if (!fqn.startsWith(`"`) && !fqn.startsWith(`'`)) {
-        // Non-quoted FQNs mean the name is a global symbol (e.g. from namespace).
-        return this.stripClutzNamespace(fqn);
-      } else {
-        // TODO(martinprobst): Quoted FQNs mean the name is from a module. We still need to scope it
-        // for ambient declarations in externs.
+    if (useFqn && this.isForExterns) {
+      // For regular type emit, we can use TypeScript's naming rules, as they match Closure's name
+      // scoping rules. However when emitting externs files for ambients, naming rules change. As
+      // Closure doesn't support externs modules, all names must be global and use global fully
+      // qualified names. The code below uses TypeScript to convert a symbol to a full qualified
+      // name and then emits that.
+      let fqn = this.typeChecker.getFullyQualifiedName(sym);
+      if (fqn.startsWith(`"`) || fqn.startsWith(`'`)) {
+        // Quoted FQNs mean the name is from a module, e.g. `'path/to/module'.some.qualified.Name`.
+        // tsickle generally re-scopes names in modules that are moved to externs into the global
+        // namespace. That does not quite match TS' semantics where ambient types from modules are
+        // local. However value declarations that are local to modules but not defined do not make
+        // sense if not global, e.g. "declare class X {}; new X();" cannot work unless `X` is
+        // actually a global.
+        // So this code strips the module path from the type and uses the FQN as a global.
+        fqn = fqn.replace(/^["'][^"']+['"]\./, '');
+      }
+      // Declarations in module can re-open global types using "declare global { ... }". The fqn
+      // then contains the prefix "global." here. As we're mapping to global types, just strip the
+      // prefix.
+      const isInGlobal = (sym.declarations || []).some((d: ts.Node | undefined) => {
```

Comment:
Why the |undefined here?  If it's only needed for the loop, it'd be clearer to make a new local var than to rely on the function type coercing.

```
(node: ts.Node) => {
  let d: ts.Node|undefined = node;
...
````

otherwise if my guess is wrong I don't get this code.